### ISSUE TITLE
Armsmaster modifier is broken in the API

### ISSUE THREAD

**Issue by ttgmpsn** [Bungie-net/api#676]
Hey! The Armsmaster modifier only exists once in the DestinyActivityModifierDefinition manifest - as 3362074814 with the description listing Anything, Fusion, Fusion as the allowed weapons. In the game (and IIRC, in the Manifest before Forsaken released), there are many different Armsmaster combinations, all with different descriptions.

The current prestige raid lairs include the modifierHash 3362074814 when queried via GetCharacter, besides actually having another Armsmaster ("Armsmaster: Hand Cannon, Sniper, Anything") active this week. Looks like modifiers with the same name got merged along the way somewhere ;)

**Comment by vthornheart-bng** [comment]
Hmm, interesting - how about the "Loadouts" data I'm returning in Milestones: is that the data you're looking for potentially?

... if it is, I could see moving this data/making this data also available on the "Activities" data so it's not just for Milestone-driven activities.  Let me know, and I can see about making that happen.

**Comment by vthornheart-bng** [comment]
Ah, yucky.  I think I see what's going on.  On the game side, I think they're dynamically generating those Armsmaster strings from the Loadout requirements data, and this string we're getting for the modifier is some placeholder they threw in that is replaced in-game by the actual loadout data determined at runtime.

... not fun there if that is the case.  I'll have to do some more investigating, but if that's the situation I might have to do something to either make the Armsmaster description more generic and rely on you to combine it with Loadout data, or alter the way Modifier data is generated in our own definitions to return every possible combination.  That more robust solution is unfortunately a bit more risky, so my gut is that I'll probably do the first solution now, and if we ever have time try to do the latter solution.

**Comment by vthornheart-bng** [comment]
TFS 710660

**Comment by sundevour** [comment]
any info on whether this is fixed/still wrong? my results are strange.

this week's prestige raid milestones point to `3704166961` which the manifest identifies as "Anything, Fusion, Fusion"
this actually **is** this week's armsmaster loadout in-game, which if it's broken, is confusing or an extreme coincidence.

however the above user says `3362074814` is "Anything, Fusion, Fusion" (now that hash is void singe).
also a coincidence?

there's only one armsmaster in the manifest i pulled: `3704166961`
i'd think the manifest was maybe returning only the current week's loadout, but this copy was fetched last week, before weekly reset, and is correct for this week.

**Comment by bhollis** [comment]
OK, I did manage to dig through the `loadouts` property on `DestinyActivityDefinition` via the `loadoutRequirementIndex` on the `DestinyMilestoneChallengeActivity`, which does return the correct loadout info. So I think the correct information is available, just not through the Modifier.

**Comment by vthornheart-bng** [comment]
Yes, unfortunately the modifier system and the loadout requirement systems are a bit different from each other, which can definitely cause confusion!

I still need to look into is whether the modifier system is returning false positive info.

**Comment by mrphlip** [comment]
It appears that at some point the loadout information disappeared from the prestige lair milestones... the `loadoutRequirementIndex` on the milestone seems to be gone, and the `loadouts` list in the manifest is empty...

**Comment by vthornheart-bng** [comment]
Hmm, I will file a new bug on this, that's definitely new.
